internal/alb/lb/attributes.go

```diff
@@ -0,0 +1,204 @@
+	a = &Attributes{
+		DeletionProtectionEnabled: DeletionProtectionEnabled,
+		AccessLogsS3Enabled:       AccessLogsS3Enabled,
+		AccessLogsS3Bucket:        AccessLogsS3Bucket,
+		AccessLogsS3Prefix:        AccessLogsS3Prefix,
+		IdleTimeoutTimeoutSeconds: IdleTimeoutTimeoutSeconds,
+		RoutingHTTP2Enabled:       RoutingHTTP2Enabled,
+	}
+	for _, attr := range attrs {
+		switch aws.StringValue(attr.Key) {
+		case DeletionProtectionEnabledString:
+			a.DeletionProtectionEnabled, err = strconv.ParseBool(aws.StringValue(attr.Value))
+			if err != nil {
+				return a, fmt.Errorf("invalid load balancer attribute value %s=%s", aws.StringValue(attr.Key), aws.StringValue(attr.Value))
+			}
+		case AccessLogsS3EnabledString:
+			a.AccessLogsS3Enabled, err = strconv.ParseBool(aws.StringValue(attr.Value))
+			if err != nil {
+				return a, fmt.Errorf("invalid load balancer attribute value %s=%s", aws.StringValue(attr.Key), aws.StringValue(attr.Value))
+			}
+		case AccessLogsS3BucketString:
+			a.AccessLogsS3Bucket = aws.StringValue(attr.Value)
+		case AccessLogsS3PrefixString:
+			a.AccessLogsS3Prefix = aws.StringValue(attr.Value)
+		case IdleTimeoutTimeoutSecondsString:
+			a.IdleTimeoutTimeoutSeconds, err = strconv.ParseInt(aws.StringValue(attr.Value), 10, 64)
+			if err != nil {
+				return a, fmt.Errorf("invalid load balancer attribute value %s=%s", aws.StringValue(attr.Key), aws.StringValue(attr.Value))
+			}
+			if a.IdleTimeoutTimeoutSeconds < 1 || a.IdleTimeoutTimeoutSeconds > 4000 {
+				return a, fmt.Errorf("%s must be within 1-4000 seconds", aws.StringValue(attr.Key))
+			}
+		case RoutingHTTP2EnabledString:
+			a.RoutingHTTP2Enabled, err = strconv.ParseBool(aws.StringValue(attr.Value))
+			if err != nil {
+				return a, fmt.Errorf("invalid load balancer attribute value %s=%s", aws.StringValue(attr.Key), aws.StringValue(attr.Value))
+			}
+		default:
+			return a, fmt.Errorf("invalid load balancer attribute %s", aws.StringValue(attr.Key))
+		}
+	}
+	return a, nil
+}
+
+// AttributesController provides functionality to manage Attributes
+type AttributesController interface {
+	// Reconcile ensures the load balancer attributes in AWS matches the state specified by the ingress configuration.
+	Reconcile(context.Context, *Attributes) error
+}
+
+// NewAttributesController constructs a new attributes controller
+func NewAttributesController(elbv2 elbv2iface.ELBV2API) AttributesController {
+	return &attributesController{
+		elbv2: elbv2,
+	}
+}
+
+type attributesController struct {
+	elbv2 elbv2iface.ELBV2API
+}
+
+func (c *attributesController) Reconcile(ctx context.Context, desired *Attributes) error {
+	raw, err := c.elbv2.DescribeLoadBalancerAttributes(&elbv2.DescribeLoadBalancerAttributesInput{
+		LoadBalancerArn: aws.String(desired.LbArn),
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed to retrieve attributes from ELBV2 in AWS: %s", err.Error())
+	}
+
+	current, _ := NewAttributes(raw.Attributes)
```

Comment:
ignored error isn't good enough
suppose a default new attribute get added in aws and appears first in attribute list.
All the attributes we get in current will be default inited, and will cause we always modify attributes over and over...